classes/ObjectModel.php

```diff
@@ -1468,7 +1468,13 @@ public function getCachedFieldsRequiredDatabase($all = false)
     {
         $this->cacheFieldsRequiredDatabase($all);
 
-        return $all ? self::$fieldsRequiredDatabase : self::$fieldsRequiredDatabase[$this->getObjectName()];
+        if ((bool)$all) {
+            return self::$fieldsRequiredDatabase;
+        }
+
+        return !empty(self::$fieldsRequiredDatabase[$this->getObjectName()])
+            ? self::$fieldsRequiredDatabase[$this->getObjectName()]
+            : array();
```

Comment:
I'd have to test `!empty()` before storing the var, and one more time to determine the return. 
I created `$objectName` instead